VTUL/iawa
# LIBTD-1524: Code consolidation from the hyrax2 repo into iawa

* **JIRA Ticket**: https://webapps.es.vt.edu/jira/browse/LIBTD-1524
* Other Relevant Links (Meeting note, project page, related pull requests, etc.) - Batch import code was migrated from: https://github.com/VTUL/hyrax2/pull/13

# What does this Pull Request do? 
This Pull Request merges work from the `VTUL/hyrax2` repository into `VTUL/iawa`, in an effort to consolidate code bases. 

# What are the changes? 

* Migrates the batch import work from the `hyrax2` (https://github.com/VTUL/hyrax2/pull/13) repository into `iawa`. Minor tweaks were made to avoid a `NameError` in production.
* Minor code changes were made to allow the controlled vocabulary modification capability to be exposed from the Hyrax 2.1.0 admin dashboard. 

# How should this be tested?

* Test functionality using the steps in this related PR: https://github.com/VTUL/hyrax2/pull/13
* Also, check that the admin dashboard allows the admin user to view and modify the Controlled Vocabulary.

<img width="253" alt="screen shot 2018-09-21 at 2 19 48 pm" src="https://user-images.githubusercontent.com/20912658/45899000-b5156100-bda9-11e8-8387-d693595b5d2a.png">


# Additional Notes:
* Use branch `LIBTD-1524`
* When setting up the data for testing, ensure the permissions are set appropriately. Example commands from @tingtingjh and @pmather:

   ```
   # Change directories to 550
   find <starting-dir> -type d -print0 | xargs -0 chmod 550
   # Change files to 440
   find <starting-dir> -type f -print0 | xargs -0 chmod 440
   ```

* Because of open issue LIBTD-1343 (https://webapps.es.vt.edu/jira/browse/LIBTD-1343), please test this in the Rails production environment.

# Interested parties
@tingtingjh
